src/code/z_scene.c

```diff
@@ -86,197 +75,208 @@ void Scene_ReloadUnloadedObjects(SceneContext* sceneCtxt) {
                     status->id = 0;
                 } else {
                     osCreateMesgQueue(&status->loadQueue, &status->loadMsg, 1);
-                    DmaMgr_SendRequestImpl(&status->dmaReq, status->segment, objectFile->vromStart,
-                                            size, 0, &status->loadQueue, NULL);
+                    DmaMgr_SendRequestImpl(&status->dmaReq, status->segment, objectFile->vromStart, size, 0,
+                                           &status->loadQueue, NULL);
                 }
             } else if (!osRecvMesg(&status->loadQueue, NULL, OS_MESG_NOBLOCK)) {
                 status->id = id;
             }
         }
+
         status++;
     }
 }
 
-s32 Scene_FindSceneObjectIndex(SceneContext* sceneCtxt, s16 objectId) {
+s32 Object_GetIndex(ObjectContext* objectCtx, s16 objectId) {
     s32 i;
-    for(i = 0; i < sceneCtxt->objectCount; i++) {
-        if((sceneCtxt->objects[i].id < 0 ? -sceneCtxt->objects[i].id : sceneCtxt->objects[i].id) == objectId) {
+
+    for (i = 0; i < objectCtx->num; i++) {
+        if ((objectCtx->status[i].id < 0 ? -objectCtx->status[i].id : objectCtx->status[i].id) == objectId) {
             return i;
         }
     }
+
     return -1;
 }
 
-s32 Scene_IsObjectLoaded(SceneContext* actorShape, s32 index) {
-    if (actorShape->objects[index].id > 0) {
-        return 1;
+s32 Object_IsLoaded(ObjectContext* objectCtx, s32 index) {
+    if (objectCtx->status[index].id > 0) {
+        return true;
     } else {
-        return 0;
+        return false;
     }
 }
 
-void Scene_DmaAllObjects(SceneContext* sceneCtxt) {
+void Object_LoadAll(ObjectContext* objectCtx) {
     s32 i;
     s32 id;
     u32 vromSize;
 
-    for (i = 0; i < sceneCtxt->objectCount; i++) {
-        id = sceneCtxt->objects[i].id;
+    for (i = 0; i < objectCtx->num; i++) {
+        id = objectCtx->status[i].id;
         vromSize = objectFileTable[id].vromEnd - objectFileTable[id].vromStart;
 
         if (vromSize == 0) {
             continue;
         }
 
-        DmaMgr_SendRequest0(sceneCtxt->objects[i].segment, objectFileTable[id].vromStart, vromSize);
+        DmaMgr_SendRequest0(objectCtx->status[i].segment, objectFileTable[id].vromStart, vromSize);
     }
 }
 
-void* func_8012F73C(SceneContext* sceneCtxt, s32 iParm2, s16 id) {
+void* func_8012F73C(ObjectContext* objectCtx, s32 iParm2, s16 id) {
     u32 addr;
     u32 vromSize;
-    ObjectFileTableEntry* fileTableEntry;
+    RomFile* fileTableEntry;
 
-    sceneCtxt->objects[iParm2].id = -id;
-    sceneCtxt->objects[iParm2].dmaReq.vromAddr = 0;
+    objectCtx->status[iParm2].id = -id;
+    objectCtx->status[iParm2].dmaReq.vromAddr = 0;
 
     fileTableEntry = &objectFileTable[id];
     vromSize = fileTableEntry->vromEnd - fileTableEntry->vromStart;
-    // TODO: UB to cast void to u32
```

Comment:
Let's not remove the avoid UB comments.

Reply:
Isn't this sort of an issue throughout the entire repo though? Seems rather redundant 🤔

Reply:
We're going to be fixing these pointers sooner rather than later in MM. I would like to keep the comments as a guide for us to look where to fix.